execution/aggregate/khashaggregate.go

```diff
@@ -91,12 +91,16 @@ func (a *kAggregate) Next(ctx context.Context) ([]model.StepVector, error) {
 	if err != nil {
 		return nil, err
 	}
-	for i := range a.params {
-		a.params[i] = math.NaN()
-		if i < len(args) {
-			a.params[i] = args[i].Samples[0]
-			a.paramOp.GetPool().PutStepVector(args[i])
+	for i := range args {
+		val := args[i].Samples[0]
+		if val > math.MaxInt64 || val < math.MinInt64 {
+			return nil, errors.Newf("scalar value %v overflows int64", val)
 		}
+		if int(val) == 0 {
+			return nil, nil
+		}
+		a.params[i] = val
+		a.paramOp.GetPool().PutStepVector(args[i])
```

Comment:
Should we do this on the error paths too?

Reply:
Probably a good idea to always do it.

Reply:
refactored slightly so that we always return args[i] back to the pool